app/src/main/java/com/github/bsideup/liiklus/service/ReactorLiiklusServiceImpl.java

```diff
@@ -226,17 +226,31 @@
     public Mono<Empty> ack(Mono<AckRequest> request) {
         return request
                 .flatMap(ack -> {
-                    val subscription = subscriptions.get(ack.getAssignment().getSessionId());
+                    String topic;
+                    GroupId groupId;
+                    int partition;
 
-                    if (subscription == null) {
-                        log.warn("Subscription is null, returning empty Publisher. Request: {}", ack.toString().replace("\n", "\\n"));
-                        return Mono.empty();
+                    if (ack.hasAssignment()) {
+                        val subscription = subscriptions.get(ack.getAssignment().getSessionId());
+
+                        if (subscription == null) {
+                            log.warn("Subscription is null, returning empty Publisher. Request: {}", ack.toString().replace("\n", "\\n"));
+                            return Mono.empty();
+                        }
+
+                        topic = subscription.getTopic();
```

Comment:
WDYT about multiple returns instead of assigning this vars each time?

Reply:
any really good reason for it?

Reply:
less lines, 
more obvious path when you stop processing at some point and just return current values. 
`- else`